integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/lowlevel/reader/LuceneIndexSegmentFilesIT.java

```diff
@@ -0,0 +1,236 @@
+			assertThat( next.hits() ).hasSize( 5 );
+
+			IndexWorkspace workspace = index.createWorkspace();
+			workspace.mergeSegments( OperationSubmitter.blocking(), UnsupportedOperationBehavior.FAIL ).join();
+
+			// we've merged the segments, so we expect that there's exactly 1 current segment
+			assertThat( numberOfCurrentSegments() ).isEqualTo( 1 );
+			// we do have a scroll opened so there are some stale files:
+			assertThat( numberOfAllSegments() ).isGreaterThan( 1 );
+			// scroll still works:
+			next = scroll.next();
+			assertThat( next.hits() ).hasSize( 5 );
+
+			// now we are going to close the scroll releasing any stale files:
+			scroll.close();
+			assertThat( numberOfCurrentSegments() ).isEqualTo( 1 );
+			Awaitility.await()
+					.timeout( 10, TimeUnit.SECONDS )
+					.untilAsserted( () -> assertThat( numberOfAllSegments() ).isEqualTo( 1 ) );
+		} );
+	}
+
+	@Test
+	void purge() throws Exception {
+
+		basicTestSteps( () -> {
+			// now let's start scrolling so that we are going to lock some of the to-be-stale segments:
+			SearchScroll<DocumentReference> scroll = index.query().where( f -> f.matchAll() ).scroll( 5 );
+			SearchScrollResult<DocumentReference> next = scroll.next();
+			assertThat( next.hits() ).hasSize( 5 );
+
+			IndexWorkspace workspace = index.createWorkspace();
+			workspace.purge( Set.of(), OperationSubmitter.blocking(), UnsupportedOperationBehavior.FAIL ).join();
+
+			// we've purged the segments, so we expect that there's exactly 0 current segment
+			assertThat( numberOfCurrentSegments() ).isZero();
+			// we do have a scroll opened so there are some stale files:
+			assertThat( numberOfAllSegments() ).isGreaterThan( 1 );
+			// scroll still works:
+			next = scroll.next();
+			assertThat( next.hits() ).hasSize( 5 );
+
+			// now we are going to close the scroll releasing any stale files:
+			scroll.close();
+			assertThat( numberOfCurrentSegments() ).isZero();
+			Awaitility.await()
+					.timeout( 10, TimeUnit.SECONDS )
+					.untilAsserted( () -> assertThat( numberOfAllSegments() ).isZero() );
+		} );
+	}
+
+	@Test
+	void noReaderMerge() throws Exception {
+		noReader( workspace -> workspace.mergeSegments( OperationSubmitter.blocking(), UnsupportedOperationBehavior.FAIL )
+				.join() );
+	}
+
+	@Test
+	void noReaderRefresh() throws Exception {
+		noReader( workspace -> workspace.refresh( OperationSubmitter.blocking(), UnsupportedOperationBehavior.FAIL ).join() );
+	}
+
+	void noReader(Consumer<IndexWorkspace> operation) throws Exception {
+		basicTestSteps( () -> {
+			LuceneIndexManagerImpl luceneIndexManager = index.unwrapForTests( LuceneIndexManagerImpl.class );
+			assertThat( luceneIndexManager.getShardsForTests() ).hasSize( 1 );
+			Shard shard = luceneIndexManager.getShardsForTests().get( 0 );
+
+			IndexReader indexReaderBefore = shard.indexAccessorForTests().getCurrentReaderForTests();
+			assertThat( indexReaderBefore ).isNotNull();
```

Comment:
Maybe also check that `indexReaderBefore` is open? You could check that `.getRefCount() > 0`.